tests/test_scalars.py

```diff
@@ -0,0 +1,235 @@
+from __future__ import annotations
+
+import datetime as dt
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    cast,
+)
+
+import numpy as np
+import pandas as pd
+from typing_extensions import assert_type
+
+from pandas._libs.tslibs import (
+    BaseOffset,
+    NaTType,
+)
+
+if TYPE_CHECKING:
+    from pandas.core.series import (
+        PeriodSeries,
+        TimedeltaSeries,
+    )
+
+    from pandas._typing import np_ndarray_bool
+else:
+    PeriodSeries = TimedeltaSeries = np_ndarray_bool = Any
```

Comment:
delete the `else` part, and then just refer to `PeriodSeries` and `TimedeltaSeries` in quotes so it works at runtime.

See `test_timefuncs.py` for an example.

Reply:
It is a lot nicer to get intellisense on these things and to no have to noqa F401 them.

Reply:
So you can do:
```python
np_ndarray_bool = npt.NDArray[np.bool_]
PeriodSeries: TypeAlias = pd.Series
TimedeltaSeries: TypeAlias = pd.Series
```
instead and get those same benefits.

Reply:
I put these in the `else` bit but I can't remove the `if TYPE_CHECKING` since the true `TimedeltaSeries` is needed, not `Series[Any]`.  

The simplest solution would be to just make these object so they are defined without any imports

Reply:
I'm going to resolve but I think just `object` would be cleaner.